### ISSUE TITLE
[subset] OS/2 Unicode range bits shouldn't be set if unset in original

### ISSUE THREAD

**Issue by blueshade7** [harfbuzz/harfbuzz#1167]
**Problem:**
hb-subset overwrites the Unicode range bits in the OS/2 table with the range bits of Unicode characters selected for the subset font, mismatching the fonttools’ behavior where range bits are set if they are also set in the original font.

**Actual** (hb-subset result):
```
% hb-subset --output-file=hb-os2.otf --unicodes=41,100,1FC,FEFF SourceSansPro-Regular.otf
% spot -tOS/2 hb-os2.otf
### [OS/2] (0001be90)
```
_SNIP_
```
unicodeRange1     =0000000d   (Basic Latin
                               Latin Extended-A
                               Latin Extended-B)
unicodeRange2     =00000000
unicodeRange3     =00000008   (Arabic Presentation Forms-B)
unicodeRange4     =00000000
```
_SNIP_

**Expected** (fonttools result):
```
% pyftsubset --output-file=ft-os2.otf --unicodes=41,100,1FC,FEFF SourceSansPro-Regular.otf
% spot -tOS/2 ft-os2.otf
### [OS/2] (000003bc)
```
_SNIP_
```
unicodeRange1     =00000005   (Basic Latin
                               Latin Extended-A)
unicodeRange2     =00000000
unicodeRange3     =00000000
unicodeRange4     =00000000
```
_SNIP_

**Original Font:**
```
spot -tOS/2 SourceSansPro-Regular.otf
### [OS/2] (00000150)
```
_SNIP_
```
unicodeRange1     =600002f7   (Basic Latin
                               Latin-1 Supplement
                               Latin Extended-A
                               IPA Extensions
                               Spacing Modifier Letters
                               Combining Diacritical Marks
                               Basic Greek
                               Cyrillic
                               Latin Extend Additional
                               Greek Extended)
unicodeRange2     =02000001   (Superscripts and Subscripts
                               Surrogates)
unicodeRange3     =00000000
unicodeRange4     =00000000
```
_SNIP_

**Comment by behdad** [comment]
@garretrieger Can you look into this?

**Comment by blueshade7** [comment]
I am going to fix this at the same time as #1719 since manually patching fontTools results is a pain every time I regenerate subset baselines (there's still cmap diffs though).

**Comment by blueshade7** [comment]
fixed by https://github.com/harfbuzz/harfbuzz/commit/82d4bfb8f3ac30fecce41f8abe59f58ad64ba98a
